### ISSUE TITLE
Bulk tag editing

### ISSUE THREAD

**Issue by aminya** [astralapp/astral#338]
It would be nice to have a button to select all or multiple repositories and tag them accordingly.

For example, searching for 'atom', and then selecting all the appeared repos and tagging them as `atom`.

**Comment by Zodarr** [comment]
I second this. You can select multiple with CTRL+click or SHIFT+click, but cannot do anything with them. As a relative new user to the app, this would help mainly for the initaial tagging of projects, since I have some 470+ starred projects (and growing, as I find more awesome open source software).

**Comment by syropian** [comment]
You should be able to drag multiple starred items onto a single tag, but yes multi-tagging could definitely still be improved!

**Comment by Zodarr** [comment]
I didn't know You could drag&drop to existing tags, thank You. It would be nice though, that when I want to create a new tag with single/multiple selected (and drag&drop), I could drag it to "Add a tag...", and it would pop up a dialog, then create and assign the new tag to the selected projects).
Regarding the possibility of thousands of writes to the DB, I think the concern is somewhat valid. Could You make something along the line of timing the writes/send it in smaller batches (with a % indicator of some sort, so the user would know if his/her changes are saved) if it's e.g. more than 50 projects to tag? Just an idea, just asking.

**Comment by syropian** [comment]
@Zodarr Yup, there's definitely ways to circumvent mass writes, it just comes down to what solution feels good in the context of the UI. 

Will definitely give this some thought, and I should be able to start on it once I knock out a couple more important things I've been meaning to do, notably a bunch of dependency upgrades, and cache overflow issues for people with huge amounts of stars.

**Comment by aminya** [comment]
For me, a `Select all checkmark` will go a long way. 

Possibly adding an `Edit Tag` button to edit the tag for all the selected entries would be useful.


![image](https://user-images.githubusercontent.com/16418197/112153105-f40e1280-8bb0-11eb-9e43-a78a359a6077.png)

**Comment by syropian** [comment]
@aminya Like I said, that functionality isn't difficult, but a bit dangerous without further considerations — there are people that have 6,000+ stars, and this wouldn't prevent someone from mass-tagging that many stars at once which could could some serious strain on the DB.
